multi_partner_learning.py

```diff
@@ -85,23 +87,31 @@ def compute_test_score_for_single_partner(self, partner):
 
         # Train model
         logger.info("   Training model...")
-        history = model.fit(
-            partner.x_train,
-            partner.y_train,
-            batch_size=partner.batch_size,
-            epochs=self.epoch_count,
-            verbose=0,
-            validation_data=self.val_data,
-            callbacks=cb,
-        )
+        if isinstance(model, type(LogisticRegression())):
+            history = model.fit(partner.x_train, partner.y_train)
+        else:
+            history = model.fit(
+                partner.x_train,
+                partner.y_train,
+                batch_size=partner.batch_size,
+                epochs=self.epoch_count,
+                verbose=0,
+                validation_data=self.val_data,
+                callbacks=cb,
+            )
```

Comment:
Ok, just one comment for the whole `multi_partner_learning.py`:

- Wow, that's a lot, well done @aygalic!
- What is really particularly interesting I think it to work with an SKL model instead of a Keras model. It raises questions: can we make the library compatible/agnostic with different ML backends?
- The `if... else...` everywhere is not ideal. Could you find some factorization for that, a bit like with the example of `collaborative_round_fit()`?